### PR TITLE
Buoyancy Offset with rospy.topics

### PR DESCRIPTION
**Pull request by epicgamer17**
## Related Issue(s)
  - Issue #1
  - Issue #2

## Other Dependent PRs
  - PR #2
  - PR #3

## Changes Made
- Describe the changes made in this PR.

## Rationale/Other Solutions Considered
- Explain the rationale behind this PR.
- Mention any alternative solutions considered (if any).

## Tests Done
- Describe the tests that have been performed.
- Mention any relevant test cases or scenarios.

## Optional Screen Recording
Attach a screen recording of test run(s) (if applicable).

## Required Wiki Changes (if any)
- List any changes that need to be made to the project's wiki (if applicable).

## Required install_dependencies.sh Changes (if any)
- List any new dependencies that need to be installed by the install_dependencies.sh script

## Related Branches (other than one that will be merged)
- List any related branches (if applicable).

### CODE REVIEW

**Comment by MrMondrian** [comment]
@epicgamer17 did you test this? the integration tests are failing

**Comment by antoinedang** [comment]
@epicgamer17 whats the status on this PR? been open for a super long time, what's taking so long?

File: catkin_ws/src/controls/src/servers/state_quaternion_pid_server.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import rospy
 from servers.base_server import BaseServer
-from std_msgs.msg import Bool
+from std_msgs.msg import Bool, String
```
Comment (antoinedang) [inline catkin_ws/src/controls/src/servers/state_quaternion_pid_server.py:5]: why is the String import needed?

File: catkin_ws/src/controls/src/servers/state_quaternion_pid_server.py [hunk 2]
```diff
@@ -50,9 +52,14 @@ def callback(self, goal):
         my_goal = self.goal_id
         self.cancelled = False
         self.goal = goal
+        
         if self.pose is not None:
             goal_position = [self.goal.pose.position.x, self.goal.pose.position.y, self.goal.pose.position.z]
             goal_quat = np.quaternion(self.goal.pose.orientation.w, self.goal.pose.orientation.x, self.goal.pose.orientation.y, self.goal.pose.orientation.z)
+
+            self.buoyant_force_pub.publish(Bool(True))
```
Comment (antoinedang) [inline catkin_ws/src/controls/src/servers/state_quaternion_pid_server.py:60]: code looks good, I'd just reverse the logic for safety (buoyant force is off when a pose is received, and only turned on if there is a Z position goal which is below the z threshold). I think the way it is written now, even if the goal does not contain a Z position target (for example it just sets the Y position or the quaternion), the buoyancy force will activate, even though the AUV's Z position might not be below the threshold.

**Review by antoinedang — CHANGES_REQUESTED** [review]
see comments

**Pull request closed without merging**